extractor/src/main/java/org/schabi/newpipe/extractor/services/youtube/extractors/YoutubeStreamExtractor.java

```diff
@@ -503,23 +508,44 @@ public String getHlsUrl() throws ParsingException {
     @Override
     public List<AudioStream> getAudioStreams() throws ExtractionException {
         assertPageFetched();
-        final List<AudioStream> audioStreams = new ArrayList<>();
-        final YoutubeThrottlingDecrypter throttlingDecrypter = new YoutubeThrottlingDecrypter(getId());
 
-        try {
-            for (final Map.Entry<String, ItagItem> entry : getItags(ADAPTIVE_FORMATS,
-                    ItagItem.ItagType.AUDIO).entrySet()) {
-                final ItagItem itag = entry.getValue();
-                String url = entry.getKey();
-                url = throttlingDecrypter.apply(url);
-
-                final AudioStream audioStream = new AudioStream(url, itag);
-                if (!Stream.containSimilarStream(audioStream, audioStreams)) {
-                    audioStreams.add(audioStream);
+        if (audioStreams == null) {
+            audioStreams = new ArrayList<>();
+
+            try {
+                for (final ItagInfo itagInfo : getItags(ADAPTIVE_FORMATS,
+                        ItagItem.ItagType.AUDIO)) {
+                    final ItagItem itag = itagInfo.getItagItem();
+                    final String content = itagInfo.getContent();
+                    final boolean isUrl = itagInfo.getIsUrl();
+                    final AudioStream audioStream;
+
+                    if (streamType == StreamType.VIDEO_STREAM) {
```

Comment:
Wait, what? Is it normal that you check for video streams when getting the results from `getItags(AUDIO)`?

Reply:
I think you misunderstood the stream type of a YouTube content with Itag types when you saw this code. I am only checking for the stream type of the content, in order to add the correct `AudioStream` object to the `AudioStream` list.

Reply:
Oh, yeah, sorry, I misunderstood ;-)